### ISSUE TITLE
console is not showing the field if the same name field is found in multiple CRDs of the operator.

### ISSUE THREAD

**Issue by prafull01** [openshift/console#7868]
I have an operator which have multiple CRDs and some of the CRDs have the field with same name.
For example: I have a Target CRD in `spec` of it, I have `type` field. 
Same way, I have an Backup CRD in `spec` of it, I have `type` field.

Now In the CSV, I provide the specDescriptor of target type and backup type but it is showing only for the target type.

```
 - description: Represents Triliovault Target destination to store a backup.
        displayName: Target
        kind: Target
        name: targets.triliovault.trilio.io
        resources:
          - kind: Pod
            name: ''
            version: v1
        specDescriptors:
          - description: Type is the type of target for backup storage.
            displayName: Type
            path: type
            x-descriptors:
              - 'urn:alm:descriptor:com.tectonic.ui:select:NFS'
              - 'urn:alm:descriptor:com.tectonic.ui:select:ObjectStore'
```
Now according to above descriptor I should see the `type` field in spec, which I able to see as per following image:
![Screenshot from 2021-01-19 15-52-36](https://user-images.githubusercontent.com/26801029/105021382-7cb3e980-5a6e-11eb-9b52-ca7e70e63aba.png)

```
- description: Represents Triliovault Backup of any Kubernetes BackupPlan.
        displayName: Backup
        kind: Backup
        name: backups.triliovault.trilio.io
        resources:
          - kind: Job
            name: ''
            version: v1
          - kind: Pod
            name: ''
            version: v1
        specDescriptors:
          - description: Type of Backup to be taken.
            displayName: Backup Type
            path: type
            x-descriptors:
              - 'urn:alm:descriptor:com.tectonic.ui:select:Full'
              - 'urn:alm:descriptor:com.tectonic.ui:select:Incremental'
              - 'urn:alm:descriptor:com.tectonic.ui:select:Mixed'
```

However, this different CRD is not able to show the `type`, because it is previously shown with the same name in different CRD, as per the following pic:

![Screenshot from 2021-01-19 15-55-18](https://user-images.githubusercontent.com/26801029/105021640-c270b200-5a6e-11eb-9ea2-d4433496bd76.png)

/kind bug

**Comment by spadgett** [comment]
Have you confirmed that it's definitely because there are two fields with the same name and not another problem? You could check by removing the first description from the CSV YAML for instance.

We probably will need the complete CSV and CRD YAML to debug.

/assign @TheRealJon

**Comment by TheRealJon** [comment]
@prafull01 I looked into this and the cause of this issue is the `allOf` property defined on the `type` field in the backups.triliovault.trilio.io CRD. Schema's with allOf, anyOf, or oneOf properties are not supported by our form generator yet. At first glance it seems like the `allOf` property on that field is unnecessary though. Since this is a string field, an enum property defining all the valid values should be sufficient. I tried this in my testing and it fixed the issue.
